### PR TITLE
feat: add header to app

### PR DESCRIPTION
**Pull request by DorMosheZarka**
This PR introduces a unified Header component that encapsulates the Logo and future components. It's designed to be imported into the App component, providing a cohesive and semantically structured top section of the application.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,27 +1,15 @@
 import styles from './App.module.scss';
+import Header from './components/Header';
+// import HeroSection from './components/HeroSection';
+// import OptionButtons from './components/OptionButtons';
```
Comment (noammu) [inline src/App.tsx]: Please remove all comments as this PR is ready for review, and there should not be any comments except for special cases, as we talked about :)

File: src/App.tsx [hunk 2]
```diff
@@ -1,27 +1,15 @@
...
-                    justifyContent: 'center',
-                }}
-            >
-                This project is using <img src="/src/assets/vite.svg" width="12" />+
-                <img src="/src/assets/typescript.svg" width="12" />
-                Visit vitejs.dev to learn more.
-            </p>
+            <Header />
+            {/* <HeroSection />
+            <OptionButtons /> */}
```
Comment (noammu) [inline src/App.tsx]: Same about comments.

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
Comment (noammu) [inline src/components/Header/Header.tsx]: You don't need to import React here.

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import styles from './Header.module.css';
+
```
Comment (noammu) [inline src/components/Header/Header.tsx]: Why the empty line between the imports? Please put them together

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import Logo from '../Logo';
+// import NavBar from '../NavBar';
+// import UserAuth from '../UserAuth';
```
Comment (noammu) [inline src/components/Header/Header.tsx]: Comments :)

File: src/components/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+function Header() {
+    return (
+        <header className={styles.header}>
+            <Logo />
+            {/* <NavBar /> */}
```
Comment (noammu) [inline src/components/Header/Header.tsx]: Comments

File: src/components/Header/index.tsx [hunk 7]
```diff
@@ -0,0 +1,2 @@
+// src/components/Header/index.tsx
```
Comment (noammu) [inline src/components/Header/index.tsx]: Again, no need for the comment

File: src/components/Header/index.tsx [hunk 8]
```diff
@@ -0,0 +1,2 @@
...
+export { default } from './Header';
```
Comment (noammu) [inline src/components/Header/index.tsx:1]: Read about the difference between-
```export { default } from './Header';```
and
```export * from './Header';```
and then talk with me after.

File: src/components/Logo/Logo.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (noammu) [inline src/components/Logo/Logo.tsx]: Again, no need for it.
Check if your VS-Code is showing you un-used imports so you will be able to identify them easily and delete them.

File: src/index.css [hunk 10]
```diff
@@ -1,51 +1,11 @@
...
-  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
-  line-height: 1.5;
-  font-weight: 400;
-  font-synthesis: none;
-  text-rendering: optimizeLegibility;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-  -webkit-text-size-adjust: 100%;
-}
+@import url('https://fonts.googleapis.com/css2?family=Quicksand:wght@300..700&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap');
```
Comment (noammu) [inline src/index.css]: Did you use this family-font that you imported?

File: src/index.css [hunk 11]
```diff
@@ -1,51 +1,11 @@
...
-  cursor: pointer;
-  transition: border-color 0.25s;
-}
-button:hover {
-  border-color: #646cff;
-}
-button:focus,
-button:focus-visible {
-  outline: 4px auto -webkit-focus-ring-color;
+  color: black;
```
Comment (noammu) [inline src/index.css]: Check out if the color on default is already black or not (probably it is ;) ).
Also under :root, same thing.

File: src/index.css [hunk 12]
```diff
@@ -1,51 +1,11 @@
...
-a {
-  font-weight: 500;
-  color: #646cff;
-  text-decoration: inherit;
-}
-a:hover {
-  color: #535bf2;
+:root {
+  color: black;
+  background-color: #ffffff;
```
Comment (noammu) [inline src/index.css]: Isn't the default already is white? If so, lets delete it. :)
And also, in the future, lets use ```white``` or any named color so the styling will be more readable (where you can of course).

File: src/index.css [hunk 13]
```diff
@@ -1,51 +1,11 @@
...
+  background-image: repeating-linear-gradient(0deg, transparent, transparent 1px, rgba(0, 0, 0, 0.02) 1px, rgba(0, 0, 0, 0.02) 2px);
```
Comment (noammu) [inline src/index.css:2]: What is this doing? I cannot see any difference when I delete it.

File: src/App.module.scss [hunk 14]
```diff
@@ -1,9 +1,3 @@
 .App {
-  margin: 0 auto;
-  padding: 2.3em 3em 2em;
-  text-align: center;
-  display: flex;
-  flex-direction: column;
-  border: 2px dashed #888888;
-  border-radius: 30px;
+  color: black;
```
Comment (noammu) [inline src/App.module.scss]: Do we need it? What happens if we delete it?

**Review by noammu — CHANGES_REQUESTED** [review]
Don't worry about the amount of comments, most of them are small.
Good job!
BTW, ```NavBar, and UserAuth components``` are not part of this PR as I see, so remove them from the description :)

File: src/index.css (hunk 13 above)
Comment (DorMosheZarka) [reply src/index.css:2]: this creates a paper like background. its very subtle.

File: src/components/Header/index.tsx (hunk 8 above)
Comment (DorMosheZarka) [reply src/components/Header/index.tsx:1]: done.

**Review by noammu — CHANGES_REQUESTED** [review]
Please delete all the unused files in the assets directory :)

**Review by noammu — APPROVED** [review]
Good job! Good scope for the PR.

**Pull request merged**